homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Sensor to read Proxmox VE data."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import DATA_GIBIBYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import COORDINATORS, DOMAIN, PROXMOX_CLIENTS, ProxmoxEntity
+
+BYTE_TO_GIBIBYTE = 1.074e9
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None,
+) -> None:
+    """Set up sensors."""
+
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    for host_config in discovery_info["config"][DOMAIN]:
+        host_name = host_config["host"]
+        host_name_coordinators = hass.data[DOMAIN][COORDINATORS][host_name]
+
+        if hass.data[PROXMOX_CLIENTS][host_name] is None:
+            continue
+
+        for node_config in host_config["nodes"]:
+            node_name = node_config["node"]
+
+            for dev_id in node_config["vms"] + node_config["containers"]:
+                coordinator = host_name_coordinators[node_name][dev_id]
+
+                # unfound case
+                if (coordinator_data := coordinator.data) is None:
+                    continue
+
+                name = coordinator_data.name
+
+                sensors += [
+                    ProxmoxSensor(
```

Comment:
Since you are planning to add more sensors in the future, I think you should look into `SensorEntityDescription` right away.